app/services/foreman/plugin/role_lock.rb

```diff
@@ -40,8 +40,12 @@ def process_role(name, permissions, description = '')
 
       def create_plugin_role(name, permissions, description = '')
         Role.ignore_locking do
-          role = Role.create! :name => name, :origin => @plugin_id, :description => description
-          role.add_permissions!(permissions)
+          begin
+            role = Role.create! :name => name, :origin => @plugin_id, :description => description
```

Comment:
Isn't there something like a get_or_create that also does this?

Reply:
we do still want to fail hard if the creation fails for other unexpected reasons

Reply:
Fair enough. I do wonder if you want to verify the permissions are correct if the role already exists. Can't really think it through so leaving it as an exercise :)

Reply:
That could still be in progress of updating, so we may get false positives here.